Examples/Io/NuclearInteractions/src/RootNuclearInteractionParametersWriter.cpp

```diff
@@ -450,20 +446,17 @@ ActsExamples::RootNuclearInteractionParametersWriter::endRun() {
     ACTS_DEBUG("Parametrisation of final state kinematics for hard " +
                std::to_string(i) + " particle(s) final state finished");
   }
-  delete (multiplicity.first);
-  delete (multiplicity.second);
-
   gDirectory->cd();
-  tf.Write();
-  tf.Close();
-
+  tf->Write();
+  tf->Close();
```

Comment:
isn't deleting all "delete"s and not getting rid of this heap TFile tf creating memory leaks?

Reply:
Root doesn't write the objects to file right away (I don't know why). Therefore I cannot delete this objects.
This may leak (there could be some hidden Root memory management though) but since it just produces the parametrisation, this entire class is isolated from any other workflow, i.e. after the `endRun()` call, the program terminates.